### PR TITLE
Kernel and kernel config update

### LINKED ISSUE #2061: Use modules for hardware drivers
With module loading now working, we should move some device drivers to be compiled as modules.

As a rule of a thumb, we should only compile in device which are commonly used on Hypervisors (mostly virtio and hyper-v) and move all other device drivers to be compiled as modules.

This should speed up boot times on local hypervisors where we care most about these things

### PR DESCRIPTION
**Pull request by rn**
- Fix kernel dockerfile to error on `apk add` (see #)
- Update kernels to 4.11.6/4.9.33/4.4.73
- Move network drivers, networking options, NFS, and others from kernel to modules (resolves #2061)

image

### CODE REVIEW

**Comment by rn** [comment]
I've tested this with 4.11 and it removes a bunch of the network interfaces and boots on hyperkit without requiring a module load. We still have the `tunl0` and `ip6tnl0`interface because I couldn't completely move the tunnel support into modules, primarily because I left VXLAN to be compiled in

File: kernel/Dockerfile [hunk 1]
```diff
@@ -26,8 +25,7 @@ RUN apk add \
     tar \
     xz \
     xz-dev \
-    zlib-dev \
-    || true
```
Comment (ijc) [inline kernel/Dockerfile]: 👍

File: pkg/getty/profile [hunk 2]
```diff
@@ -0,0 +1,4 @@
+echo "You are running inside a container!"
+echo "You can access the host root filesystem under /proc/1/root"
+
+PS1="getty \$ "
```
Comment (ijc) [inline pkg/getty/profile]: I suspect you didn't intend to include this.

Comment (rn) [inline pkg/getty/profile]: Oops, yes. should have done a `git clean`

File: kernel/Dockerfile (hunk 1 above)
Comment (justincormack) [inline kernel/Dockerfile]: the `|| true` is there so  that you don't get whitespace changes when changing the last item or adding a new one ta the end.

Comment (rn) [inline kernel/Dockerfile]: but it also masks errors

Comment (justincormack) [inline kernel/Dockerfile]: ah, ok yes

**Comment by rn** [comment]
Hmm failed on an unrelated GCP issue. I've compiled/pushed/signed the kernel images and hit a rebuild

**Comment by rn** [comment]
kernel tests needed updating as well to remove NFS and rpc_pipefs as they have been moved to modules. Force pushed an update

**Comment by rn** [comment]
@tych0 I hadn't had a chance to update the kernel-config project, but might be good to track these changes once merged

**Comment by justincormack** [comment]
Please don't remove these from the tests, they should be there. You can add a set of things to `modprobe` before running the tests.

**Comment by justincormack** [comment]
(or just add a script to modprobe everything pre-test)

**Comment by rn** [comment]
Added the modprobe for selected modules and the test passes locally.

@justincormack if you are happy with it and the test is green, please merge as I'm off soon

File: test/hack/test.yml [hunk 3]
```diff
@@ -12,7 +12,7 @@ onboot:
     image: "linuxkit/dhcpcd:7d2b8aaaf20c24ad7d11a5ea2ea5b4a80dc966f1"
     command: ["/sbin/dhcpcd", "--nobackground", "-f", "/dhcpcd.conf", "-1"]
   - name: check-kernel-config
-    image: "linuxkit/test-kernel-config:2acaa564c1801dd2ae1546c70c472dc58ac030a1"
+    image: "linuxkit/test-kernel-config:9f08e3b99f8ac2f422251b3e8c94ce874ee34119"
     readonly: true
```
Comment (justincormack) [inline test/hack/test.yml]: did you mean to delete this `readonly: true` too?

**Comment by justincormack** [comment]
ok, one tiny comment but looks fine

File: test/hack/test.yml (hunk 3 above)
Comment (rn) [inline test/hack/test.yml]: yes, missed that one, thanks for spotting

**Comment by justincormack** [comment]
the `run_efi` test is also failing here, I don't think its related but unclear why it started failing recently (or exactly when).

**Comment by tych0** [comment]
@rn yes, I can take a look at that. We should probably also update kernel-config for the user mode helper changes.

File: kernel/Dockerfile (hunk 1 above)
Comment (ijc) [inline kernel/Dockerfile]: If we keep them sorted then the chances of something coming after `zlib` are at least slim...

Or the list could end
```
    zlib-dev \
    # end of list
```
although I'm not sure that is nicer than the occasional whitespace-fu on update.

Comment (ijc) [inline kernel/Dockerfile]: Either `&& true` or `|| false` would also give the desired result, wouldn't it?

It looks like (from `git grep -B1 '\(||\|&&\) \(true\|false\)'`) a lot of places following this sort of pattern are using the `&& true` variant.

There are other uses of `|| true`, many of which look suspect to me. Lots of

* `rm -rf ${STUFF}* || true`. But `rm -f` ignores non-existent arguments which I think is the main reason for the use of `|| true`, so it is unneeded.
* `RUN tar cf «something» usr || true`, I can't imagine why we would want to ignore such errors, we should insist on an explanatory comment for such constructs IMHO.

**Comment by justincormack** [comment]
CI failure is the ISO EFI test which needs more investigation, see #2100

**Pull request merged**